datasetinsights/io/tracker/mzflow.py

```diff
@@ -0,0 +1,84 @@
+import os
+
+import mlflow
+from google.auth.transport.requests import Request
+from google.oauth2 import id_token
+
+MLFLOW = "mlflow"
+CLIENT_ID = "tracking_client_id"
+HOST_ID = "tracking_host_id"
+NAME = "experiment_name"
+DEFAULT_NAME = "datasetinsights"
+
+
+class MLFlowTracker:
+    """MlFlow tracker class, responsible for creating singleton object,
+    refresh token and get instance .
+
+    Attributes:
+        __instance: holds singleton instance
+        __client_id: holds ml tracking server client id
+
+    """
+
+    __instance = None
+    __client_id = None
```

Comment:
these will probably become instance attributes initialized in the CTOR (i.e. `self.client_id`) now that the tracker will hold the thread that refreshes its credentials behind the scenes